### PR TITLE
Refactor course list and empty state

### PR DESCRIPTION
**Pull request by MaxFrank13**
This refactors the `CourseList` component so that all of the logic for rendering the list of `CourseCard` components is encapsulated in its own file. The new file is now called `CourseList` which lives in the new `CoursesPanel` component. The `CoursesPanel` component is identical to what the `CourseList` component used to be — there are no functional/behavior changes here.

This was done to enable better modulation with the Learner Dashboard UI through the use of the [frontend-plugin-framework](https://github.com/openedx/frontend-plugin-framework).

### CODE REVIEW

File: src/containers/CoursesPanel/index.test.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+jest.mock('containers/CourseCard', () => 'CourseCard');
+jest.mock('containers/CourseFilterControls', () => ({
+  ActiveCourseFilters: 'ActiveCourseFilters',
+  CourseFilterControls: 'CourseFilterControls',
+}));
+jest.mock('./CourseList', () => 'CourseList');
+
+reduxHooks.useHasCourses.mockReturnValue(true);
+
+describe('CourseList', () => {
```
Comment (jsnwesson) [inline src/containers/CoursesPanel/index.test.jsx:24]: Should this be to describe `CoursesPanel`?

**Review by deborahgu — APPROVED** [review]
looks good to me

**Comment by jsnwesson** [comment]
Everything looks pretty good to me! The only thing I'd request is that we end up [losing some tests in this PR](https://github.com/openedx/frontend-app-learner-dashboard/blob/e045932e5f91547fa8c84b0e61823730f9457470/src/containers/CourseList/index.test.jsx#L54-L80) that relate to how `CoursesPanel` should look like with filters and courses/no courses. Could we bring those back in somehow? Otherwise we have less eyes on the filter having any effect on the `CourseList`, no? My line of thinking could be totally wrong though, so I'm open to any opinions on this.

**Comment by MaxFrank13** [comment]
> Everything looks pretty good to me! The only thing I'd request is that we end up [losing some tests in this PR](https://github.com/openedx/frontend-app-learner-dashboard/blob/e045932e5f91547fa8c84b0e61823730f9457470/src/containers/CourseList/index.test.jsx#L54-L80) that relate to how `CoursesPanel` should look like with filters and courses/no courses. Could we bring those back in somehow? Otherwise we have less eyes on the filter having any effect on the `CourseList`, no? My line of thinking could be totally wrong though, so I'm open to any opinions on this.

@jsnwesson Good call out! I actually did move these tests to the `CourseList/index.test.js` file [here](https://github.com/openedx/frontend-app-learner-dashboard/blob/b49592c0708fa3316c09962d7dc97528d9df8b67/src/containers/CoursesPanel/CourseList/index.test.jsx#L35-L63). They aren't appearing in the changes tab for this PR and I'm not 100% sure why. One thought is that because the name of this file is the same as the previous one (just located in a different directory), source control already "knows" about those lines of code? 

Either way, feel free to take a look at that file and let me know if you still think I missed something 👍

File: src/containers/CoursesPanel/index.test.jsx (hunk 1 above)
Comment (MaxFrank13) [reply src/containers/CoursesPanel/index.test.jsx:24]: Yes! Good catch!

**Review by jsnwesson — APPROVED** [review]

**Pull request merged**